fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexExpressions.java

```diff
@@ -114,14 +117,23 @@ public static class DocumentFieldDerivation {
         private final List<String> recordFieldPath;
         @Nonnull
         private final DocumentFieldType type;
+        // TRUE when we have overridden the field name and replaced the original field with another
+        private final boolean fieldNameOverride;
+        // The replaced field name suffix (this has replaced the original in the documentField)
```

Comment:
Once the field derivation has been created, the intent of `namedFieldSuffix` is to represent the "fully realized" replaced portion of the key. 
So in our example, where the Index key expression is 
```
field("mapUidToLabel")
...
    nest(function("Lucene_field_name",
        field("value"),
        field("key").nest(field"value"))
```
the `namedFieldSuffix` is expected to be `key_value` so that the `mapUidToLabel_key_value` gets replaced with `mapUidToLabel_mylabel`

I haven't tested the nesting map example as above, but following up on another comment revealed a bug where one of the tests failed to properly match an expression of the form 
```
    nest(function("Lucene_field_name",
        field("value").nest("value"),
        field("key"))
```
(this is in the test that had a `StringToLongWrapper` structure) 
After consulting with @ScottDugas we decided to leave that out of the current support matrix (I'll open an issue) and so I removed that test.